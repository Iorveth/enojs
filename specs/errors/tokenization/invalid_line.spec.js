const eno = require('../../../eno.js');

describe('tokenization.invalidLine', () => {
  const input = `
    languages:
    - eno
    - json
    yaml
  `;

  let error;

  beforeAll(() => {
    try {
      eno.parse(input);
    } catch(err) {
      error = err;
    }
  });

  it(`provides a correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toEqual([[5, 0], [5, 8]]);
  });
});
