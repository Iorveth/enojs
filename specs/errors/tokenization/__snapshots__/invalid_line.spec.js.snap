// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenization.invalidLine provides a correct message 1`] = `
"Line 5 does not follow any specified pattern.

   Line | Content
   ...
      3 |     - eno
      4 |     - json
 >    5 |     yaml
      6 |   
"
`;

exports[`tokenization.invalidLine provides correct selection metadata 1`] = `
Array [
  Array [
    4,
    0,
  ],
  Array [
    4,
    8,
  ],
]
`;
